### ISSUE TITLE
ListUsers filter does not work

### ISSUE THREAD

**Issue by codyburrito** [jagregory/cognito-local#313]
Attempting a search for users using a filter returns the pool as if no filter was added.  Confirmed by attempting the same query in cognito local vs actual cognito

## Steps to recreate
```bash
aws --endpoint http://localhost:9229 cognito-idp list-users --user-pool-id <local user pool> --filter "family_name = \"WhateverValueYouWant\"
```

### Expected
```bash
{
  "Users": []
}
```

### Actual

```bash
{
  "Users": [...every user in pool]
}
```

**Comment by jagregory** [comment]
This behaviour is documented in the readme:

> ¹ does not support pagination or query filters, all results and attributes will be returned in the first request.

Pull Requests are welcome, this isn't a high priority issue for me personally.

**Comment by jagregory** [comment]
Typically as soon as I closed this issue saying I don't need it I immediately needed it myself 🤦 

This has been implemented and released in v3.23.0.
